### PR TITLE
Star type subscription

### PR DESCRIPTION
**Pull request by brijeshnec**

### CODE REVIEW

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java [hunk 1]
```diff
@@ -578,8 +600,11 @@ && shouldSendOut(potentialSub, message.getPayload())) {
 			case AppConstants.DELETE_ATTRIBUTE_REQUEST -> {
 
 				if (shouldFire(Sets.newHashSet(message.getAttribName()), potentialSub)) {
-					unis.add(localEntityService.getAllByIds(message.getTenant(), message.getId(), true).onItem()
+					unis.add(localEntityService.getAllByIds(message.getTenant(), message.getId(), true,potentialSub.getSubscription().isLocalOnly()).onItem()
```
Comment (amitrnec) [inline SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:603]: Add postman test cases for subscription local only functionality.

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java [hunk 2]
```diff
@@ -1649,8 +1661,9 @@ public Uni<Map<String, Object>> getEntityFlat(String tenant, String entityId, Ma
 		});
 	}
 
-	public Uni<Map<String, Object>> getEntityInline(String tenant, String entityId, Map<String, Object> attrsMap,
-			String containedBy, int joinLevel, Context context) {
+public Uni<Map<String, Object>> getEntityInline(Context context, String tenant, String entityId, Map<String, Object> attrsMap,
```
Comment (amitrnec) [inline QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java:1664]: Check and fix indentation.

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java [hunk 3]
```diff
@@ -167,6 +183,9 @@ public static Subscription expandSubscription(Map<String, Object> body, String i
 						case NGSIConstants.JSON_LD_TYPE:
 							hasType = true;
 							entityInfo.setType(((List<String>) entitiesEntry.getValue()).get(0));
+							if(entityInfo.getType().equals("https://uri.etsi.org/ngsi-ld/default-context/*")){
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: Add in NGSIConstants "https://uri.etsi.org/ngsi-ld/default-context/*"

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java [hunk 4]
```diff
@@ -123,6 +133,12 @@ public static Subscription expandSubscription(Map<String, Object> body, String i
 			Object mapValue = mapEntry.getValue();
 
 			switch (key) {
+				case NGSIConstants.NGSI_LD_LOCALONLY:
+					if(!(mapValue instanceof List<?> list && ((Map<String,Object>)(list.get(0))).get("@value") instanceof Boolean)){
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: You can take it from NGSIConstants "@value"

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java [hunk 5]
```diff
@@ -123,6 +133,12 @@ public static Subscription expandSubscription(Map<String, Object> body, String i
...
+						throw new ResponseException(ErrorType.BadRequestData, "localOnly should be boolean");
+					}
+					subscription.setLocalOnly(((List<Map<String,Boolean>>)mapValue).get(0).get("@value"));
```
Comment (amitrnec) [inline Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: Same as here "@value"

File: SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java (hunk 1 above)
Comment (vivekNEC) [reply SubscriptionManager/src/main/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionService.java:603]: done

File: QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java (hunk 2 above)
Comment (vivekNEC) [reply QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java:1664]: done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java (hunk 3 above)
Comment (vivekNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java (hunk 4 above)
Comment (vivekNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: done

File: Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java (hunk 5 above)
Comment (vivekNEC) [reply Commons/src/main/java/eu/neclab/ngsildbroker/commons/datatypes/Subscription.java]: done

**Review by ScorpioBroker — APPROVED** [review]

**Pull request merged**